### PR TITLE
server: Add txn check

### PR DESCRIPTION
**Pull request by zimulala**
If the `txn` is invalid, we needn't rollback it.

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -654,8 +654,10 @@ func (cc *clientConn) handleLoadData(loadDataInfo *executor.LoadDataInfo) error
 
 	txn := loadDataInfo.Ctx.Txn()
 	if err != nil {
-		if err1 := txn.Rollback(); err1 != nil {
-			log.Errorf("load data rollback failed: %v", err1)
+		if txn.Valid() {
```
Comment (shenli) [inline server/conn.go]: When will txn be invalid?

Comment (zimulala) [inline server/conn.go]: The committing is failed in the function of `insertDataWithCommit`.

**Comment by zimulala** [comment]
PTAL @shenli @coocood @tiancaiamao @hanfei1991

**Comment by hanfei1991** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**